### PR TITLE
gitignore: Ignore vim files

### PR DESCRIPTION
**Pull request by sirspinach**
Also, remove some extraneous comments at the top of the file.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,3 @@
-## Ignore Visual Studio temporary files, build results, and
-## files generated by popular Visual Studio add-ons.
```
Comment (apanda) [inline .gitignore]: Is there a reason to remove this comment. It is largely true.

Comment (sirspinach) [inline .gitignore]: Oh, I thought it was supposed to be a section header (like "User-specific files")

Comment (sirspinach) [inline .gitignore]: I'll change this back

Comment (sirspinach) [inline .gitignore]: @apanda: fixed

**Pull request merged**